### ISSUE TITLE
Use JAX sparse module for numerical impedance predictions

### ISSUE THREAD

**Issue by BradyPlanden** [pybop-team/PyBOP#596]
### Feature description

Add a Jax sparse solving method for the numeric impedance prediction to enable gradient-based optimisers.

### Motivation

Currently, our implementation of numerical impedance uses the numpy sparse solver `spsolve`. Jax has experimental functionality for sparse matrices and operations (see [here](https://jax.readthedocs.io/en/latest/jax.experimental.sparse.html#module-jax.experimental.sparse)) with functionality for `jax.grad` and similar functions.
